### PR TITLE
fix: Decoupling importing segment from flush process

### PR DESCRIPTION
**Pull request by bigsheeper**
This pr decoups importing segment from flush process by:
1. Exclude the importing segment from the flush policy, this approch avoids notifying the datanode to flush the importing segment, which may not exist.
2. When RootCoord call Flush, DataCoord directly set the importing segment state to `Flushed`.

issue: https://github.com/milvus-io/milvus/issues/30359

### CODE REVIEW

File: internal/datacoord/segment_manager.go [hunk 1]
```diff
@@ -481,8 +481,17 @@ func (s *SegmentManager) SealAllSegments(ctx context.Context, collectionID Uniqu
 			ret = append(ret, id)
 			continue
 		}
-		// segment can be sealed only if it is growing or if it's importing
-		if (!isImport && info.State != commonpb.SegmentState_Growing) || (isImport && info.State != commonpb.SegmentState_Importing) {
+		// Decoupling the importing segment from the flush process.
+		// Hence, we seal and flush the importing segment without returning it.
+		// This approach prevents the flush validation of this importing segment.
+		if isImport && info.State == commonpb.SegmentState_Importing {
+			if err := s.meta.SetState(id, commonpb.SegmentState_Flushed); err != nil {
```
Comment (xiaofan-luan) [inline internal/datacoord/segment_manager.go:488]: why we set this as flushed?
what if impor failed?

**Review by xiaofan-luan — COMMENTED** [review]
Question:

GetFlushableSegments call, we do we simply remove segment in bulkinsert states
because they are definitely not flushable

**Comment by bigsheeper** [comment]
rerun ut

**Comment by bigsheeper** [comment]
rerun ut

**Comment by bigsheeper** [comment]
rerun ut

**Pull request merged**